design/batching.md

```diff
@@ -0,0 +1,1018 @@
+# End-to-end batching
+
```

Comment:
Since this will end up in the repository it's probably a good idea to add a date and potentially an author